src/devices/CharacterLcd/Hd44780.cs

```diff
@@ -68,181 +77,312 @@ public class Hd44780 : Hd44780Base, IDisposable
+        /// 
+        /// Only 6 bits of addresses are available for character ram. That makes for 64 bytes of
+        /// available character data. 8 bytes of data are used for each character, which is where
+        /// the 8 total custom characters comes from (64/8).
+        /// 
+        /// Each byte corresponds to a character line. Characters are only 5 bits wide so only
+        /// bits 0-4 are used for display. Whatever is in bits 5-7 is just ignored. Store bits
+        /// there if it makes you happy, but it won't impact the display. '1' is on, '0' is off.
+        /// 
+        /// In the built-in characters the 8th byte is usually empty as this is where the underline
+        /// cursor will be if enabled. You can put data there if you like, which gives you the full
+        /// 5x8 character. The underline cursor just turns on the entire bottom row.
+        /// 
+        /// 5x10 mode is effectively useless as displays aren't available that utilize it. In 5x10
+        /// mode *16* bytes of data are used for each character. That leaves room for only *4*
+        /// custom characters. The first character is addressable from code 0, 1, 8, and 9. The
+        /// second is 2, 3, 10, 11 and so on...
+        /// 
+        /// In this mode *11* bytes of data are actually used for the character data, which
+        /// effectively gives you a 5x11 character, although typically the last line is blank to
+        /// leave room for the underline cursor. Why the modes are referred to as 5x8 and 5x10 as
+        /// opposed to 5x7 and 5x10 or 5x8 and 5x11 is a mystery. In an early pre-release data
+        /// book 5x7 and 5x10 is used (Advance Copy #AP4 from July 1985). Perhaps it was a
+        /// marketing change?
+        /// 
+        /// As only 11 bytes are used in 5x10 mode, but 16 bytes are reserved, the last 5 bytes
+        /// are useless. The datasheet helpfully suggests that you can store your own data there.
+        /// The same would be true for bits 5-7 of lines that matter for both 5x8 and 5x10.
+        /// </remarks>
+        /// <param name="location">Should be between 0 and 7</param>
+        /// <param name="characterMap">Provide an array of 8 bytes containing the pattern</param>
+        public void CreateCustomCharacter(byte location, params byte[] characterMap)
+        {
+            if (location > 7)
+                throw new ArgumentOutOfRangeException(nameof(location));
+
+            if (characterMap.Length != 8)
+                throw new ArgumentException(nameof(characterMap));
+
+            SendCommand((byte)(SetCGRamAddressCommand | (location << 3)));
+            SendData(characterMap);
         }
 
-        private void WriteBits(byte value, int count)
+        /// <summary>
+        /// Write text to display.
+        /// </summary>
+        /// <remarks>
+        /// There are only 256 characters available. There are chip variants
```

Comment:
This is actually interesting. One one hand you want  simple API which takes string on the other hand this is actually is a byte array which partially maps to ASCII... I think this is fine as is but wondering if just for clarity we should not also have `ReadOnlySpan<byte>` overload which describes that this is actually custom character set 8-bit encoding which mostly overlaps with ASCII

Reply:
There are a ton of character sets out there, including custom chips. I give quite a bit of detail in the remarks.  Taking a `byte` would require contortion more often than not- .NET uses `string` and `char`.  If you want to pass "raw" data you can do so, I just don't expect that to be common. When we get UTF-8 core types we should review all of our APIs that take strings.
 
Note that I'm not concerned about perf here as the character display is really quite small (80 characters at max).